src/components/labelPrinting/LabelPrintingForm.vue

```diff
@@ -0,0 +1,194 @@
+          type="number"
+          :min="1"
+          :max="10"
+          placeholder="Please select a number"
+          required
+        ></b-form-input>
+      </b-form-group>
+
+      <b-button id="resetButton" type="reset" variant="danger" class="float-left">Reset</b-button>
+
+      <labelPrintingModal
+        id="labelPrintingModal"
+        ref="labelPrintingModal"
+        class="float-right"
+        :disabled="!formValid"
+        v-bind="propsToPass()"
+      ></labelPrintingModal>
+    </b-form>
+  </div>
+</template>
+
+<script>
+import LabelPrintingModal from '@/components/labelPrinting/LabelPrintingModal'
+const MESSAGE_SELECT = 'Please select a option'
+
+export default {
+  name: 'LabelPrintingForm',
+  components: {
+    LabelPrintingModal,
+  },
+  data() {
+    return {
+      form: {
+        barcode: null,
+        selectedSuffixId: null,
+        selectedNumberOfLabels: null,
+        selectedPrinterId: null,
+        copies: null,
+      },
+      suffixOptions: [],
+      printerOptions: [],
+      show: true,
+    }
+  },
+  computed: {
+    formValid() {
+      return !!(
+        this.form.barcode &&
+        this.form.selectedSuffixId &&
+        this.form.selectedNumberOfLabels &&
+        this.form.selectedPrinterId &&
+        this.form.copies
+      )
+    },
+  },
+  created() {
+    this.setSuffixOptions()
+    this.setPrinterNames()
+  },
+  methods: {
+    setSuffixOptions() {
+      let suffixOptions = this.$store.getters.suffixes.map((obj, index) => ({
+        value: index + 1,
+        text: obj['one_character_name'] + ' (' + obj['potential_label_name'] + ')',
+        char: obj['one_character_name'],
+      }))
+      suffixOptions.unshift({ value: null, text: MESSAGE_SELECT })
+      this.suffixOptions = suffixOptions
+    },
+    setPrinterNames() {
+      let printerOptions = this.$store.getters.printersWithType.map((obj, index) => ({
+        value: index + 1,
+        text: obj.printerName,
+        type: obj.printerType,
+      }))
+      printerOptions.unshift({ value: null, text: MESSAGE_SELECT })
+      this.printerOptions = printerOptions
+    },
+    propsToPass() {
+      const props = {}
+
+      if (this.formValid) {
+        props.barcodesList = this.suffixedBarcodes()
+        props.printer = this.printer()
+        props.copies = this.form.copies
+      }
+
+      return props
+    },
+    suffixedBarcodes() {
+      var listSuffixedBarcodes = []
+
+      if (this.formValid) {
+        let suffixString = this.suffixOptions[this.form.selectedSuffixId].char
+
+        for (let i = 0; i < this.form.selectedNumberOfLabels; i++) {
+          listSuffixedBarcodes.push(this.form.barcode.concat('-', suffixString, i + 1))
+        }
+      }
+      return listSuffixedBarcodes
+    },
+    printer() {
+      return this.printerOptions[this.form.selectedPrinterId]
+    },
+    onReset(event) {
+      event.preventDefault()
+      // Reset our form values
+      this.form.barcode = null
```

Comment:
From above you could then just do `this.form = newForm` or something similar. If that works?